src/components/ScreenWrapper.tsx

```diff
@@ -160,6 +161,16 @@ function ScreenWrapper(
 
     isKeyboardShownRef.current = keyboardState?.isKeyboardShown ?? false;
 
+    const route = useRoute();
+    const shouldReturnToOldDot = useMemo(() => {
+        return !!(route?.params && 'singleNewDotEntry' in route.params && route?.params?.singleNewDotEntry === 'true');
+    }, [route]);
+
+    UNSTABLE_usePreventRemove(shouldReturnToOldDot, () => {
+        Navigation.setParams({singleNewDotEntry: undefined});
+        NativeModules.HybridAppModule.closeReactNativeApp(false, false);
+    });
+
```

Comment:
To clarify for reviewers, it is stable 😄 The same code is present in the react-navigation v7 just without `UNSTABLE` prefix.

However, this is an argument for considering upgrading to v7. We see other use cases where features from v7 would also be useful.